### PR TITLE
Changes related to PA-RTF merge

### PR DESCRIPTION
**Pull request by atharvasharma61**
**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Issue details to be added

**Describe the solution you are proposing**
Introducing support to read telemetry-collectors-enabled flag value from performance-analyzer.properties

**Describe alternatives you've considered**

**Additional context**

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/performanceanalyzer/commons/collectors/TelemetryCollector.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.commons.collectors;
+
+public interface TelemetryCollector {}
```
Comment (Gaganjuneja) [inline src/main/java/org/opensearch/performanceanalyzer/commons/collectors/TelemetryCollector.java:8]: java doc please.

File: src/main/java/org/opensearch/performanceanalyzer/commons/collectors/ScheduledMetricCollectorsExecutor.java [hunk 2]
```diff
@@ -159,4 +176,13 @@ public void run() {
             }
         }
     }
+
+    private boolean canSchedule(PerformanceAnalyzerMetricsCollector collector) {
+        if (collector instanceof TelemetryCollector) {
+            return (collectorsSetting == Util.CollectorMode.DUAL.getValue())
+                    || (collectorsSetting == Util.CollectorMode.OTEL.getValue());
```
Comment (Gaganjuneja) [inline src/main/java/org/opensearch/performanceanalyzer/commons/collectors/ScheduledMetricCollectorsExecutor.java:183]: Can we please keep the names in sync like here OTEL should be replaced with telemetry or RTF.

File: src/main/java/org/opensearch/performanceanalyzer/commons/util/Util.java [hunk 3]
```diff
@@ -85,4 +85,20 @@ public static void invokePrivilegedAndLogError(Runnable runner) {
                             return null;
                         });
     }
+
+    public enum CollectorMode {
```
Comment (Gaganjuneja) [inline src/main/java/org/opensearch/performanceanalyzer/commons/util/Util.java:89]: java doc please.

**Review by Gaganjuneja — CHANGES_REQUESTED** [review]

**Review by Gaganjuneja — DISMISSED** [review]

File: src/main/java/org/opensearch/performanceanalyzer/commons/collectors/ScheduledMetricCollectorsExecutor.java [hunk 4]
```diff
@@ -16,6 +16,7 @@
 import org.apache.logging.log4j.Logger;
 import org.opensearch.performanceanalyzer.commons.stats.ServiceMetrics;
 import org.opensearch.performanceanalyzer.commons.stats.metrics.StatMetrics;
+import org.opensearch.performanceanalyzer.commons.util.Util;
 
 public class ScheduledMetricCollectorsExecutor extends Thread {
```
Comment (nishchay21) [inline src/main/java/org/opensearch/performanceanalyzer/commons/collectors/ScheduledMetricCollectorsExecutor.java:21]: lets add documentation on classes where missing during PR

**Review by nishchay21 — APPROVED** [review]

**Review by Gaganjuneja — APPROVED** [review]

**Pull request merged**